src/lib/update-lock.ts

```diff
@@ -82,6 +82,55 @@ async function dispose(
 	}
 }
 
+/**
+ * Given a lockfile path `p`, return a tuple [appId, serviceName] of that path.
+ * Paths are assumed to end in the format /:appId/:serviceName/(resin-)updates.lock.
+ */
+function getIdentifiersFromPath(p: string) {
+	const parts = p.split('/');
+	if (parts.pop()?.match(/updates\.lock/) === null) {
+		return [];
+	}
+	const serviceName = parts.pop();
+	const appId = parts.pop();
+	return [appId, serviceName];
+}
+
+export type LocksTakenMap = Map<number, string[]>;
```

Comment:
Looking at later commit tests, I feel that aliasing map might not be the best way to implement this as it makes the code hard to read.